tests/json_schema.go

```diff
@@ -24,23 +24,25 @@ import (
 	"regexp"
 	"strings"
 	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/elastic/apm-server/model/metadata"
+	"github.com/elastic/apm-server/processor/stream"
+
+	"github.com/elastic/apm-server/tests/loader"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/mapping"
 )
 
-type TestProcessor interface {
```

Comment:
While refactoring processors, I deemed necessary to actually test those real processors, as I didn't necessarily trusted the mock processors. 
As a matter of fact, testing the production code directly did catch a couple of issues.
Making the production code testable by itself without needing lots of custom logic is a good thing IMO, but in this case if forced some additional changes, like eg. https://github.com/elastic/apm-server/pull/3254#discussion_r376847765

Reply:
I see you closed this PR, but your comment above about issues you caught worries me:
> As a matter of fact, testing the production code directly did catch a couple of issues.

Please create issues for them and link them here.

Reply:
@jalvz sorry, I was unclear. I didn't mean to suggest they weren't needed at all, but rather that they could be separated into a separate PR. Because there's a lot of changes involved, it would help to review them independently. Same with the other similar comments I made.

Reply:
> I didn't mean to suggest they weren't needed at all, but rather that they could be separated into a separate PR

Sure. It just felt a bit creepy to leave the tests for another PR. Most of the changes are related to the `Transform` and `Validate` change, thou.

> your comment above about issues you caught worries me

I meant about implementation changes in this PR, not existing code from before.

Reply:
ah I misunderstood then, thanks for clarification.